module/zfs/dsl_scan.c

```diff
@@ -3356,6 +3357,25 @@ dsl_scan_sync(dsl_pool_t *dp, dmu_tx_t *tx)
 	if (spa->spa_syncing_txg < spa->spa_first_txg + SCAN_IMPORT_WAIT_TXGS)
 		return;
 
+	/*
+	 * zfs_scan_disable_progress can be set to disable scan progress.
+	 * We don't want to spin the txg_sync thread, so we add a delay
+	 * here to simulate the time spent doing a scan. This is mostly
+	 * useful for testing and debugging.
+	 */
+	if (zfs_scan_disable_progress) {
+		uint64_t scan_time_ns = gethrtime() - scn->scn_sync_start_time;
+
+		while (zfs_scan_disable_progress &&
+		    !txg_sync_waiting(scn->scn_dp) &&
+		    !spa_shutting_down(scn->scn_dp->dp_spa) &&
+		    NSEC2SEC(scan_time_ns) < zfs_txg_timeout) {
```

Comment:
I don't think we want to delay for the entire txg.  Could we instead simply delay for either the `zfs_resilver_min_time_ms`  or `zfs_scrub_min_time_ms` time as appropriate.

Reply:
I'm a bit confused. Why would we not? It's not too big of a difference either way, but this would result in shorter txgs while the flag was active, which I'm not sure we'd want.

Reply:
Wasting the `txg_sync` threads time doing absolutely nothing doesn't seem productive.  I understand having a delay so we don't spin, but it seems like if we could advance a few txgs that would be a good thing for testing for any txg waiters.  Though since this is purely for testing perhaps it's OK.